Source/Mothership/TMoth.cpp

```diff
@@ -938,6 +1021,246 @@ void TMoth::StopTwig()
 
 //------------------------------------------------------------------------------
 
+
+
+//------------------------------------------------------------------------------
+// Instrumentation Methods
+//------------------------------------------------------------------------------
+
+// Instrumentation initialisation
+//void TMoth::InstrumentationInit(void)
+//{
+//    
+//}
+
+// Gracefully tear down the Instrumentation map
+void TMoth::InstrumentationEnd(void)
+{
+    WALKMAP(uint32_t,TM_Instrumentation*,InstrMap,I)
+        delete I->second;
+}
+
+
+// Handle an instrumentation message
+unsigned TMoth::InstrumentationHandler(P_Msg_t* msg)
+{
+    TM_Instrumentation* instr;
+    uint32_t srcAddr = msg->header.pinAddr;
+    
+    // Pointer to the Instrumentation
+    P_Instr_Msg_Pyld_t* instrMsg = reinterpret_cast<P_Instr_Msg_Pyld_t*>(msg->payload);
+    
+    std::ofstream tFile;        // Thread instrumentation file
+    // Set Filename
+    std::ostringstream fName;
+    fName << "instrumentation_thread_" << srcAddr << ".csv";
+    
+    TM_InstrMap_t::iterator MSearch = InstrMap.find(srcAddr);
+    if(MSearch == InstrMap.end())
+    { // Instrumentation for a new thread.
+        // Form a new instrumentation record
+        instr = new TM_Instrumentation();
+        instr->totalTime = 0;
+        instr->txCount = 0;
+        instr->rxCount = 0;
+        
+        /* This falls over for large plates, so we open every time.
+        // Set Filename
+        std::ostringstream fName;
+        fName << "instrumentation_thread_" << srcAddr << ".csv";
+        
+        // Open file
+        instr->tFile.open(fName.str(), std::ofstream::out);
+        */
+        
+        // Open file and overwrite contents
+        tFile.open(fName.str(), std::ofstream::out);
+        
+        // Check it is open
+        if(tFile.fail())    //instr->tFile.fail()) // Check that the file opened
+        {   // if it didn't, tell logserver, delete the entry and return
+            Post(541, fName.str(), POETS::getSysErrorString(errno));
+            
+            delete instr;
+            return 1;
+        }
+        
+        // Write the CSV header
+        tFile << "ThreadID, cIDX, Time, cycles, deltaT, ";
+        tFile << "RX, OnRX, TX, SupTX, OnTX, Idle, OnIdle, Blocked, ";
+#if TinselEnablePerfCount == true
+        tFile << "CacheMiss, CacheHit, CacheWB, CPUIdle, ";
+#endif
+        tFile << "RX/s, TX/s, Sup/s" << std::endl;
+         
+        tFile << srcAddr << ", 0, 0, 0, 0, ";

```

Comment:
Why do we have this line of zeroes here? :sweet_potato:

Reply:
Ah, the comment for that has disappeared. It is to give a 0 start for everything to make some of the graphing easier.